From e725fb033a428a436aa701c9dd63fdf3c53df352 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 30 Jul 2024 10:34:34 +0300
Subject: [PATCH] [FSW-7920] fs1: recovery: remove d2d phy fsm

D2D PHY FSM registers has changed in HW tag 1.14, until the new sequence
is finalized, to support build with 1.14 specs, this functionality is
removed.

Change-Id: Ie317abcebbe60aea124974dedce3072c2188e944
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/dev_recovery.c b/arch/fs1/arc/dev_recovery.c
index f3fadf9..d607a10 100644
--- a/arch/fs1/arc/dev_recovery.c
+++ b/arch/fs1/arc/dev_recovery.c
@@ -75,9 +75,6 @@
 
 void boot_rom_d2d_link_init(void)
 {
-	uint64_t start;
-	uint32_t val;
-
 	if (boot_rom_global.sku == FS1_IO_DIE_SA)
 		return;
 
@@ -86,13 +83,7 @@
 
 	boot_rom_status_update(PROGRESS_BOOT_ROM_D2D_LINK_START);
 
-	/* TODO: IO HALF0 FSM_STS and HALF1 FSM_STS are in sync */
-	/* Check BUSY bit for 0 value, expected to be 0 */
-	val = regmap_readl(IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_DPHY_FSM_STS));
-	if (val & GLOBAL_CONF_DPHY_FSM_STS_BUSY_M) {
-		boot_rom_status_update(PROGRESS_BOOT_ROM_D2D_LINK_STS_BUSY);
-		return;
-	}
+	/* TODO: check IO HALF0 FSM_STS and HALF1 FSM_STS are in sync */
 
 	/* Master Die and Slave Die's sync for d2d rec init */
 	if (boot_rom_global.die_type == FS1_IO_DIE_MASTER) {
@@ -105,25 +96,9 @@
 	/* Inter die indication for d2d link init handshaking */
 	boot_rom_status_update(PROGRESS_BOOT_ROM_D2D_LINK_INTER_DIE_SYNC);
 
-	/* TODO: IO HALF0 FSM_CTL and HALF1 FSM_CTL are in sync */
-	/* Set START bit for D2D link initialization */
-	regmap_writel(GLOBAL_CONF_DPHY_FSM_CTRL_START_M, IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_DPHY_FSM_CTRL));
+	/* TODO: IO HALF0 FSM_CTL and HALF1 FSM_CTL are in sync, and set start bit of D2D init */
 
-	/*
-	 * Poll 10ms for BUSY bit to be 0 indicating link initialization
-	 * is completed successfully.
-	 */
-	start = boot_rom_get_ts_us();
-	while (1) {
-		val = regmap_readl(IO0_H0_PSOC_GLOBAL_CONF(mmGLOBAL_CONF_DPHY_FSM_STS));
-		if (!(val & GLOBAL_CONF_DPHY_FSM_STS_BUSY_M))
-			break;
-
-		if (boot_rom_timeout_expired(start, D2D_LINK_STATUS_UDELAY)) {
-			boot_rom_status_update(PROGRESS_BOOT_ROM_D2D_LINK_INIT_TIMEDOUT);
-			return;
-		}
-	}
+	/* TODO: wait for DPYH FSM busy status */
 
 	boot_rom_status_update(PROGRESS_BOOT_ROM_D2D_LINK_INIT_SUCCESS);
 
